main.cpp

```diff
@@ -228,346 +305,521 @@ struct University
-    //amount of daily used disk space
     float diskSpace = 89.5f;
-    //number of execution errors
     int execErrors = 48;
-    //task executed per day
     int execTask = 25;
+    bool programmingSoftwareInstalled = false;
+    bool canSaveInfo;
 
-    //Computer Hardware Specs
     struct Hardware  
     { 
-        //[Gb] Total RAM
-        float RAM = 16; 
-        //[Gb] Total Disk space
-        float disk = 1024; 
-        //[GHz] CPU Speed
-        float CPU = 4.5;
-        //[Gb] Total GPU 
-        float GPU = 16;
-        //[Inches] Screen Size
-        float screen = 24; 
-
+        double RAM = 16.0; 
+        double disk = 1024.0; 
+        double CPU = 4.5;
+        double GPU = 16.0;
+        double screen = 24.0;
+        bool canPlayGames;
+        bool canTrainAI;
+        bool canUseAtOffice;
+        
         bool playGames(bool specs = true); 
         bool trainAI(std::string softwareReq = "Keras"); 
         bool workAtOffice(std::string mostlyUsedTask = "Accounting"); 
     };
 
-    //Execute Programs 
-    void executePrograms(Computer officeDesktop,  
-    std::string todayTask = "Programming");  
-    //Save Information
+    void executePrograms(std::string todayTask = "Programming");  
     bool saveInfo(bool diskAvailable = true); 
-    //Connect with other computers
-    void connectToPCs(Computer Server, bool LANavailable = true);
+    void connectToPCs(bool LANavailable = true);
 };
 
-/*
-Thing 4) Audio Mixer
-5 properties:
-    1) number of input channels (int)
-    2) number of output channels (int)
-    3) number of signal processing options (int)
-    4) number of simultaneous processing objects (int)
-    5) speed of DSP (float)
-3 things it can do:
-    1) mix multiple signals into an stereo one
-    2) split signal to be delivered at different points
-    3) process signal (EQ, Fx, Dynamics)
- */
- 
+//Computer OfficeDesktop;
+
+bool Computer::Hardware::playGames(bool specs)
+{
+    if (specs == true && RAM >= 4 && disk >=256 && CPU >= 3.0 && GPU >= 2)
+    {
+        std::cout << "You can use this computer to play games";  
+    }
+    return canPlayGames = true;
+}
+
+bool Computer::Hardware::trainAI(std::string softwareReq)
+{
+    if (canPlayGames == true && GPU > 4)
+    {
+        std::cout << "You can use this computer to train AI using" << softwareReq;
+    }
+    return canTrainAI = true;
+}
+
+bool Computer::Hardware::workAtOffice(std::string mostlyUsedTask)
+{
+    if (canPlayGames == true)
+    {
+        std::cout << "You can use this computer to do " << mostlyUsedTask;
+    }
+    return canUseAtOffice = true;
+} 
+
+void Computer::executePrograms(std::string todayTask)
+{
+    if (programmingSoftwareInstalled == true)
+    {
+        std::cout << "This computer can be used  to" << todayTask;
+    }
+}
+
+bool Computer::saveInfo(bool diskAvailable)
+{
+    if (diskAvailable == true)
+    {
+        std::cout << "The file will be saved";
+    }
+    return canSaveInfo = true;
```

Comment:
Do not write code like this.
What value are you returning?
it is impossible to reason which value is being returned.
Separate the assignment `= true` from the value being returned.
```
canSaveInfo = true;
return .......
```
- [ ] revise